.github/workflows/release.yaml

```diff
@@ -0,0 +1,84 @@
+name: Release
+
+on:
+    push:
+        branches: [ main, master ]
+    pull_request:
+    release:
+        types: [ created ]
+
+jobs:
+    build-phar:
+        runs-on: ubuntu-latest
+        name: Build PHAR
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v3
+                with:
+                    fetch-depth: 0
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: '8.1'
+                    ini-values: phar.readonly=0
+                    tools: composer
+                    coverage: none
+
+            -   name: Install Composer dependencies
+                uses: ramsey/composer-install@v2
+
+            -   name: Build PHAR
+                run: |
+                    make phpbrew
+                    mv phpbrew bin/phpbrew.phar
+
+            # Smoke test
+            -   name: Ensure the PHAR works
+                run: bin/phpbrew.phar --version
+
+            -   name: Import GPG key
+                if: github.event_name == 'release'
+                uses: crazy-max/ghaction-import-gpg@v5
+                with:
+                    gpg_private_key: ${{ secrets.GPG_KEY_<TODO:privatekeyID> }}
+                    passphrase: ${{ secrets.GPG_KEY_<TODO:privatekeyID>_PASSPHRASE }}
+
+            -   name: Sign the PHAR
+                if: github.event_name == 'release'
+                run: |
+                    gpg --local-user <TODO:email> \
+                        --batch \
+                        --yes \
+                        --passphrase="${{ secrets.GPG_KEY_<TODO:privatekeyID>_PASSPHRASE }}" \
+                        --detach-sign \
+                        --output bin/phpbrew.phar.asc \
+                        bin/phpbrew.phar
+
+            -   uses: actions/upload-artifact@v3
+                name: Upload the PHAR artifact
+                with:
+                    name: phpbrew-phar
+                    path: |
+                        bin/phpbrew.phar
+                        bin/phpbrew.phar.asc
+
+    publish-phar:
+        runs-on: ubuntu-latest
+        name: Publish PHAR
+        needs:
+            - build-phar
+        if: github.event_name == 'release'
+        steps:
+            -   uses: actions/download-artifact@v3
+                with:
+                    name: phpbrew-phar
+                    path: .
+
+            -   name: Upload phpbrew.phar
+                uses: softprops/action-gh-release@v1
+                with:
+                    token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
@c9s, could you generate the secret GITHUB_TOKEN in this repository?

Reply:
To double check but I think this one is _already_ provided by GitHub (when not in a PR)

Reply:
You're right.

The GITHUB_TOKEN is a special access token that you can use to authenticate on behalf of GitHub Actions.

GitHub automatically creates a GITHUB_TOKEN secret for you to use in your workflow, and you can use it to authenticate in a workflow run.